charon/tests/ui/simple/multiple-promoteds.out

```diff
@@ -0,0 +1,87 @@
+# Final LLBC before serialization:
+
+pub fn core::ops::arith::{impl core::ops::arith::Add<&'_0 (u32)> for &'_1 (u32)}#27::add<'_0, '_1>(@1: &'_1 (u32), @2: &'_0 (u32)) -> u32
+
+fn test_crate::six() -> u32
+{
+    let @0: u32; // return
+    let x@1: &'_ (u32); // local
+    let y@2: &'_ (u32); // local
+    let @3: &'_ (u32); // anonymous local
+    let @4: &'_ (u32); // anonymous local
+    let @5: &'_ (u32); // anonymous local
+    let @6: &'_ (u32); // anonymous local
+    let @7: &'_ (u32); // anonymous local
+    let @8: u32; // anonymous local
+    let @9: (u32, bool); // anonymous local
+    let @10: &'_ (u32); // anonymous local
+    let @11: u32; // anonymous local
+    let @12: (u32, bool); // anonymous local
+
+    storage_live(@5)
+    storage_live(@6)
+    storage_live(@7)
+    storage_live(@8)
+    storage_live(@9)
+    storage_live(@10)
+    storage_live(@11)
+    storage_live(@12)
+    storage_live(x@1)
+    @9 := const (0 : u32) checked.+ const (1 : u32)
```

Comment:
I'm surprised that there are no checks for overflow here. Is it normal or did Charon miss something?

Reply:
I must have gotten mixed up with commits because I noticed and was intending to fix that. I think what happens is that the overflow check stays in the function but the addition is promoted, which confuses the binop reconstruction pass.